src/main/java/racingcar/domain/CarGroup.java

```diff
@@ -0,0 +1,57 @@
+package racingcar.domain;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class CarGroup {
+
+    private final List<Car> carGroup;
+
+    public static CarGroup createCarGroupWithName(List<String> names) {
+        List<Car> cars = new ArrayList<>();
+        for (String name : names) {
+            cars.add(new Car(name, 0));
+        }
+        return new CarGroup(cars);
+    }
+
+    public CarGroup(List<Car> carGroup) {
+        this.carGroup = carGroup;
+    }
+
+    public void moveAll() {
+        for (Car car : carGroup) {
+            car.move();
+        }
+    }
+
+    private int getMaxPosition() {
+        int maxPosition = 0;
+        for (Car car : carGroup) {
+            maxPosition = Math.max(maxPosition, car.getPosition());
+        }
+        return maxPosition;
+    }
+
+    public CarGroup getWinner() {
+        int maxPosition = getMaxPosition();
+        List<String> winnerNames = new ArrayList<>();
+        List<Car> winnerCars = carGroup.stream()
+                .filter((car) -> car.isMatched(maxPosition))
+                .collect(Collectors.toList());
+        return new CarGroup(winnerCars);
+    }
+
+    public List<String> getWinnerName() {
+        return this.getWinner().getCarGroup().stream()
+                .map(Car::getName)
+                .collect(Collectors.toList());
+    }
```

Comment:
여기서  getWinner() 메소드에서 CarGroup 객체를 받아서 다시 우승자이름목록을 구하는데 두번 하지 말고 합쳐도 될것 같습니다.
stream을 사용하는 이유가 여러 작업을 chaining 해서 쓸수 있기 때문에 분명 장점이 있거든요.
물론 단순 for문으로 처리가능한 로직은 for문이 더 빠르므로 적절히 사용하시면 됩니다.  
```suggestion
public List<String> getWinnerName() {
        int maxPosition = getMaxPosition();
        return carGroup.stream()
                .filter((car) -> car.isMatched(maxPosition))
                .map(Car::getName)
                .collect(Collectors.toList());
    }